src/wfs/src/main/java/org/geoserver/wfs/kvp/FilterKvpParser.java

```diff
@@ -81,7 +82,9 @@ public Object parse(String value) throws Exception {
                 InputStream input = new ByteArrayInputStream(string.getBytes());
 
                 try {
-                    Filter filter = (Filter) parser.parse(input);
+                    InputSource inputSource = new InputSource(input);
+                    inputSource.setEncoding(Charset.defaultCharset().name());
```

Comment:
Thanks @aaime, similar to your last advice about "GeoServerInfo.getSettings().getCharset()", could it be better replace the hard-coded "UTF-8" encoding in that "getEncodingName" function ?

Similar to:
https://github.com/ahuarte47/geoserver/commit/d658eab273fc61f001a3041d1ed4015378962d54#diff-17cf56a09a7f63467323d1504f02d3daR13